src/spikeinterface/core/core_tools.py

```diff
@@ -9,10 +9,55 @@
 import importlib
 from math import prod
 from collections import namedtuple
+import inspect
 
 import numpy as np
 
 
+def create_dict_of_source_classes(dict_of_recs, source_class, *args, **kwargs):
+    """Given a dict of recordings, return a dict of recordings with `source_class` applied to them."""
+    preprocessed_recordings_dict = {
+        property_id: source_class(recording, *args, **kwargs) for property_id, recording in dict_of_recs.items()
+    }
+    return preprocessed_recordings_dict
+
+
+def define_function_or_dict_from_class(source_class, name):
+    """
+    Depending on whether `source_class` is passed a `Recording` object or a dict of
+    `Recording` objects, this function will return `source_class` or a dict of
+    `source_class` objects to match the input.
+    """
+
+    from spikeinterface.core import BaseRecording
+
+    def source_class_or_dict_of_sources_classes(*args, **kwargs):
+
+        recording_in_kwargs = False
+        if rec_or_dict_of_recs := kwargs.get("recording"):
+            recording_in_kwargs = True
+        else:
+            rec_or_dict_of_recs = args[0]
+
+        if isinstance(rec_or_dict_of_recs, BaseRecording):
+            return source_class(*args, **kwargs)
+        else:
```

Comment:
I would do 
```suggestion
        elif isinstance(, dict):
```

and then raise an error when it is something else

Reply:
Yess, great idea